pkg/controller/llmisvc/controller_int_test.go

```diff
@@ -569,6 +569,273 @@ var _ = Describe("LLMInferenceService Controller", func() {
+				}, expectedSecret)
+			}).WithContext(ctx).Should(Succeed())
+
+			return expectedSecret
+		}
+
+		waitForRoleBinding := func(ctx context.Context, nsName string) *rbacv1.ClusterRoleBinding {
+			expectedClusterRoleBinding := &rbacv1.ClusterRoleBinding{}
+			Eventually(func(_ Gomega, ctx context.Context) error {
+				return envTest.Get(ctx, types.NamespacedName{
+					Name: "kserve-metrics-reader-role-binding-" + nsName,
+				}, expectedClusterRoleBinding)
+			}).WithContext(ctx).Should(Succeed())
+
+			return expectedClusterRoleBinding
+		}
+
+		waitForPodMonitor := func(ctx context.Context, nsName string) {
+			expectedPodMonitor := &monitoringv1.PodMonitor{}
+			Eventually(func(_ Gomega, ctx context.Context) error {
+				return envTest.Get(ctx, types.NamespacedName{
+					Name:      "kserve-llm-isvc-vllm-engine",
+					Namespace: nsName,
+				}, expectedPodMonitor)
+			}).WithContext(ctx).Should(Succeed())
+		}
+
+		waitForServiceMonitor := func(ctx context.Context, nsName string) *monitoringv1.ServiceMonitor {
+			expectedServiceMonitor := &monitoringv1.ServiceMonitor{}
+			Eventually(func(_ Gomega, ctx context.Context) error {
+				return envTest.Get(ctx, types.NamespacedName{
+					Name:      "kserve-llm-isvc-scheduler",
+					Namespace: nsName,
+				}, expectedServiceMonitor)
+			}).WithContext(ctx).Should(Succeed())
+
+			return expectedServiceMonitor
+		}
+
+		waitForAllMonitoringResources := func(ctx context.Context, nsName string) {
+			waitForServiceMonitor(ctx, nsName)
+			waitForSASecret(ctx, nsName)
+			waitForRoleBinding(ctx, nsName)
+			waitForPodMonitor(ctx, nsName)
+			waitForServiceMonitor(ctx, nsName)
+		}
+
+		waitConsistentlyForAllMonitoringResources := func(ctx context.Context, nsName string) {
+			expectedServiceAccount := &corev1.ServiceAccount{}
+			expectedSecret := &corev1.Secret{}
+			expectedClusterRoleBinding := &rbacv1.ClusterRoleBinding{}
+			expectedPodMonitor := &monitoringv1.PodMonitor{}
+			expectedServiceMonitor := &monitoringv1.ServiceMonitor{}
+
+			Consistently(func(g Gomega, ctx context.Context) {
+				g.Expect(envTest.Get(ctx, types.NamespacedName{
+					Name:      "kserve-metrics-reader-sa",
+					Namespace: nsName,
+				}, expectedServiceAccount)).To(Succeed())
+
+				g.Expect(envTest.Get(ctx, types.NamespacedName{
+					Name:      "kserve-metrics-reader-sa-secret",
+					Namespace: nsName,
+				}, expectedSecret)).To(Succeed())
+
+				g.Expect(envTest.Get(ctx, types.NamespacedName{
+					Name: "kserve-metrics-reader-role-binding-" + nsName,
+				}, expectedClusterRoleBinding)).Should(Succeed())
+
+				g.Expect(envTest.Get(ctx, types.NamespacedName{
+					Name:      "kserve-llm-isvc-vllm-engine",
+					Namespace: nsName,
+				}, expectedPodMonitor)).Should(Succeed())
+
+				g.Expect(envTest.Get(ctx, types.NamespacedName{
+					Name:      "kserve-llm-isvc-scheduler",
+					Namespace: nsName,
+				}, expectedServiceMonitor)).Should(Succeed())
+			}).WithContext(ctx).Should(Succeed())
+		}
```

Comment:
This makes the tests less readable because I need to scroll down to understand the specific scenario, can we extract those functions into separate fixtures/helpers?

Reply:
+1 and lets make them accepting name/ns as params at least so we can re-use them for other tests.

Reply:
I moved to the end in 94c7ab6.